src/runtime/HalideRuntime.h

```diff
@@ -1871,48 +1868,97 @@ struct HALIDE_ATTRIBUTE_ALIGN(8) halide_profiler_pipeline_stats {
     int num_allocs;
 };
 
-/** The global state of the profiler. */
+/** Per-invocation-of-a-pipeline state. Lives on the stack of the Halide
+ * code. Exists in a doubly-linked list to that it can be cleanly
+ * removed. */
+struct halide_profiler_instance_state {
+    /** Time spent in this instance. */
+    uint64_t time;
```

Comment:
As per comments further below: take both `sampled_time` and `wall_time`.

Reply:
Will fix in next push. The thing I needed to track was the wall clock time of the pipeline start.